src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1163,6 +1177,56 @@ def _create_client_secret():
     client_secret = binascii.b2a_hex(os.urandom(10)).decode('utf-8') + special_char
     return client_secret
 
+def _ensure_aks_acr(cli_ctx,
+                    acr_name,
+                    cluster_name,
+                    location,
+                    client_id,
+                    resource_group_name,
+                    subscription_id):
+    acr_client = cf_container_registry_service(cli_ctx)
+    # Generate a default ACR name.
+    registry_name = acr_name
+    if registry_name is None or registry_name == "":
+        registry_name = "aks" + cluster_name + "acr"
```

Comment:
The generated registry_name needs more thoughts: cluster_name supports special chars (e.g. `-`) while the registry name doesn't.  We should probably remove all the special chars from cluster_name here.

Reply:
one concern here is the cluster name might be duplicated, like I always create my test cluster with name thomasaks